teek/_widgets/treeview.py

```diff
@@ -0,0 +1,545 @@
+    ::
+
+        treeview.rows[0].subrows.append(teek.TreeviewRow(values=['a', 'b'])
+
+    .. attribute:: config
+
+        Similar to the ``config`` attribute that widgets have. The available
+        options are documented as ``ITEM OPTIONS`` in :man:`ttk_treeview(3tk)`.
+    """
+    _next_row_num = 0
+
+    def __init__(self, name=None, **kwargs):
+        self._name = name or 'R%d' % TreeviewRow._next_row_num
+        self._creation_opts = kwargs
+        self._treeview = None
+        self._root = None
+
+        TreeviewRow._next_row_num += 1
+
+        self.config = OptionConfigDict(self._config_handler)
+        self.config._types.update({
+            'values': [str],
+            'text': str,
+            'open': bool,
+            'tags': [str],
+            'image': teek.Image
+        })
+
+        self.subrows = TreeviewRowList(treeview=self._treeview, root=self)
+
+    def __repr__(self):
+        repr_str = '%s(%s' % (self.__class__.__name__, self._name)
+
+        if self._treeview:
+            repr_str += ', text=%r, values=%r, subrows=%d' % (
+                self.config['text'],
+                self.config['values'],
+                len(self.subrows)
+            )
+        else:
+            repr_str += ', not added to treeview'
+
+        return repr_str + ')'
+
+    def __eq__(self, other):
+        return isinstance(other, TreeviewRow) and self._name == other._name
+
+    def _config_handler(self, rettype, option=None, *args):
+        self._check_added()
+
+        if option is not None:
+            args = [option] + list(args)
+
+        return self._treeview._call(
+            rettype, self._treeview, 'item', self, *args
+        )
+
+    def _check_added(self):
+        """
+        Check if row is added to treeview.
+        """
+        if self._treeview is None:
+            raise RuntimeError(
+                "this row hasn't been added to a treeview yet")
+
+    def _assign(self, treeview, root=None, set_config=True):
+        """
+        Internal method for assigning a treeview and a root node to current
+        row.
+        The `set_config` argument specifies if init options should be set.
+        """
+        self._treeview = treeview
+        self._root = root
+
+        self.subrows._treeview = treeview
+
+        if treeview is not None and set_config:
+            for name, value in self._creation_opts.items():
+                self.config[name] = value
+
+    @property
+    def selected(self):
+        """
+        Get current selection state.
+        """
+        self._check_added()
+        return self._name in self._treeview._call([str], self._treeview,
+                                                  'selection')
+
+    @make_thread_safe
+    def select(self):
+        """
+        Add row to current selection.
+        """
+        self._check_added()
+        self._treeview._call(None, self._treeview, 'selection', 'add', self)
+
+    @make_thread_safe
+    def deselect(self):
```

Comment:
When you name methods differently than the corresponding command in the manual page, can you mention that in the docstring? Something like "this calls `selection remove` documented in blah blah". You could also name this Python method `selection_remove`.